src/main/java/tc/oc/pgm/restart/RestartManager.java

```diff
@@ -1,171 +1,92 @@
-    return queuedAt;
-  }
-
-  public @Nullable String restartReason() {
-    return reason;
-  }
-
-  public boolean isRestartRequested() {
-    return restartRequestedAt() != null;
-  }
-
-  public boolean isRestartDeferred() {
-    return !this.deferrals.isEmpty();
-  }
-
-  public boolean isRestartDeferredBy(RequestRestartEvent.Deferral deferral) {
-    return this.deferrals.contains(deferral);
-  }
-
-  public void requestTimedRestart(Duration duration) {
-    if (queuedRestartTask > 0) {
-      this.plugin.getServer().getScheduler().cancelTask(queuedRestartTask);
-      queuedRestartTask = 0;
-    }
-    Instant queueStamp = Instant.now();
-    long ticks = duration.getStandardSeconds() * 20;
-    queuedRestartTask =
-        Bukkit.getScheduler()
-            .runTaskLater(
-                this.plugin,
-                () -> {
-                  queuedRestartTask = 0;
-                  if (!this.deferrals.isEmpty()) return;
-                  logger.info("Restarting due to request at " + queueStamp);
-                  this.plugin.getServer().shutdown();
-                },
-                ticks)
-            .getTaskId();
-  }
-
-  public void requestRestart(String reason) {
-    if (!this.isRestartRequested()) {
-      this.queuedAt = Instant.now();
-      this.reason = reason;
-    }
-  }
-
-  public void cancelRestart() {
-    if (queuedRestartTask > 0) {
-      this.plugin.getServer().getScheduler().cancelTask(queuedRestartTask);
-      queuedRestartTask = 0;
-    }
-    if (this.isRestartRequested()) {
-      this.queuedAt = null;
-      this.reason = null;
-    }
-  }
-
-  public void deferRestart(RequestRestartEvent.Deferral deferral) {
-    if (this.isRestartRequested()) {
-      this.logger.info("Restart deferred by " + deferral.getPlugin().getName());
-      this.deferrals.add(deferral);
-    }
-  }
-
-  public void resumeRestart(RequestRestartEvent.Deferral deferral) {
-    this.logger.info("Restart resumed by " + deferral.getPlugin().getName());
-    this.deferrals.remove(deferral);
-    this.restartIfRequested();
-  }
-
-  public boolean restartIfRequested() {
-    if (this.isRestartRequested() && this.deferrals.isEmpty()) {
-      this.plugin
-          .getServer()
-          .getScheduler()
-          .runTask(
-              this.plugin,
-              new Runnable() {
-                @Override
-                public void run() {
-                  logger.info("Restarting due to request at " + restartRequestedAt());
-                  plugin.getServer().shutdown();
-                }
-              });
-      return true;
-    } else {
-      return false;
-    }
-  }
-}
+package tc.oc.pgm.restart;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import tc.oc.pgm.api.event.RequestRestartEvent;
+
+public class RestartManager {

```

Comment:
How about instead of having a static `RestartManager`, change all of its methods and fields to be static.